### DISCUSSION TITLE
Using external hardware

### DISCUSSION THREAD

**Discussion by moltar** [mediar-ai/screenpipe#68] [Ideas]
Is it possible to use an external hardware box to capture and process the feed?

E.g. I don't want to load my system more than necessary, so I was thinking of using an old laptop, or even some dedicated hardware box, to capture the feed via, perhaps, HDMI or USB-C and then do the processing externally.

Would you feel like it'd provide a perf win over just using it locally?

**Comment by louis030195** [comment]
@moltar in fact someone did it, running screenpipe on a cloud machine that uses local mics & screens, will share how to do so

**Reply by louis030195** [reply]
![image](https://github.com/user-attachments/assets/80412ff8-f02c-493d-af80-6806b1069c17)

```bash
ssh -R 4713:localhost:4713 -X remote-user@remote-host bash -c "PULSE_SERVER=localhost /opt/screen-pipe/target/release/screenpipe"
```

was on a linux local & remote setup

can probably be hacked around with different local OS too, maybe you can ask `sabrehagen_` on discord how he did it more in details

**Comment by louis030195** [comment]
@moltar also would you use cloud based solution to process transcriptions and screen text?

**Reply by moltar** [reply]
I am not opposed to the cloud on a privacy principle.

My main issue is that my internet is unstable; thus, I wanted to use a locally networked machine.

**Comment by louis030195** [comment]
@moltar we did some optims, screenpipe only uses 600 mb to 4 gb and 10-30% cpu on mac book pro m3, LMK any news on this :)

**Comment by thekumbhaj** [comment]
Hey @moltar  hope your work in progress,
let me know

**Comment by roxxid** [comment]
Any luck with this? I wanna run the backend and store my data as it is generated on my home server using a docker container, but I am guessing I need to install the desktop app on my machines which can record the data and stream it to the backend?
